folsom/src/main/java/com/spotify/folsom/MemcacheClientBuilder.java

```diff
@@ -406,22 +412,51 @@ public MemcacheClientBuilder(final Transcoder<V> valueTranscoder) {
     return this;
   }
 
+  /**
+   * Authenticate with memcached using plaintest SASL. This only works for binary connections, not ascii.
+   * @param username
+   * @param password
+   * @return itself
+   */
+  public MemcacheClientBuilder<V> withUsernamePassword(final String username, final String password) {
+    this.authenticator = new PlaintextAuthenticator(username, password);
+    return this;
+  }
+
+  /**
+   * Disable authentication validation - only useful for tests against
+   * jmemcached which does not support binary NOOP
+   *
+   * @return itself
+   */
+  MemcacheClientBuilder<V> withoutAuthenticationValidation() {
+    this.authenticator = NoAuthenticationValidation.INSTANCE;
+    return this;
+  }
+
+
   /**
    * Create a client that uses the binary memcache protocol.
    * @return a {@link com.spotify.folsom.BinaryMemcacheClient}
    */
   public BinaryMemcacheClient<V> connectBinary() {
+    final Authenticator authenticator = firstNonNull(
```

Comment:
Having a potentially null field seems unnecessary. How about initializing it with BinaryAuthenticationValidator.INSTANCE?

Reply:
Then the ascii case gets a lot more complicated